src/include/bootloader/routines/misc.rb

```diff
@@ -182,25 +198,28 @@ def setKernelParamToLine(line, key, value)
         k = kernel_param_key(param)
         if k != key # not our param
           res << param
-        elsif value == "false"
+        elsif values == "false"
           next # do nothing as we want to remove this param
         elsif occurences[k] == 1 # last parameter with given key
           done = true
-          if value == "true"
+          if values == "true"
             res << key
-          elsif value != "false"
-            res << Builtins.sformat("%1=%2", key, value)
+          elsif values != "false"
+            Array(values).each do |v|
+              res << Builtins.sformat("%1=%2", key, v)
```

Comment:
and now you hit your problem. With `verbose crashkernel=bla,bla verbose` read and write, then result will be `verbose=true crashkernel=bla,bla verbose=true` which is wrong.